src/utilities/dialogs/Overlay.tsx

```diff
@@ -0,0 +1,120 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { Popper } from '@components/Popper';
+import React, { Children, isValidElement, useCallback, useRef, useState } from 'react';
+import { Trigger } from '@utilities/dialogs/Trigger';
+import { Content } from '@utilities/dialogs/Content';
+import { OVERLAY_CONTAINER_CLASSES } from '@utilities/overlayStyle';
+import { BaseDialogProps, Modality, OverlayProps } from '../../types/dialog';
+import { merge } from '@utilities/merge';
+import { useFocusTrap } from '@hooks/useFocusTrap';
+import { useClickOutside } from '@hooks/useClickOutside';
+import { Portal } from '@components/Portal';
+import { useDropdownAutoHeight } from '@hooks/useDropdownAutoHeight';
+import { Z_INDEX_MODAL_BACKDROP } from '@utilities/dialogs/constants';
+
+export const Overlay = ({
+    open,
+    placement,
+    offset,
+    flip,
+    enablePortal,
+    children,
+    role = 'region',
+    'data-test-id': dataTestId = 'fondue-overlay',
+    maxWidth = 'auto',
+    maxHeight = 'auto',
+    minWidth = 0,
+    minHeight = 0,
+    handleClose,
+    modality = Modality.NonModal,
+    darkUnderlay,
+    autoHeight,
+    zIndex,
+}: OverlayProps & BaseDialogProps) => {
+    const id = useMemoizedId();
+    const ref = useRef<HTMLDivElement | null>(null);
+    const [triggerElementRef, setTriggerElementRef] = useState<HTMLDivElement | null>(null);
+    const { maxHeight: maxAutoHeight } = useDropdownAutoHeight(
+        { current: triggerElementRef },
+        { isOpen: open, autoResize: true },
+    );
+
+    const maxContentHeight = autoHeight ? maxAutoHeight : maxHeight;
+    const handleClosingInteraction = useCallback(() => {
+        if (open && modality !== Modality.BlockingModal) {
+            if (handleClose) {
+                handleClose();
+            }
+        }
+    }, [handleClose, modality, open]);
+
+    useFocusTrap(ref.current, open, modality === Modality.NonModal);
```

Comment:
In the popover story with interactive content the focus is not contained within the flyout. Is this intentional? It seems like a case when you would have the focus captured.

Reply:
Intentional, for now.